File: packages/javascript/lib/word/randProduct.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Things you buy — the manufactured end of the vocabulary.
 *
 * `randWord({ theme: 'product' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randProduct({ language: 'ko', count: 3 }); // ['노트북', '키보드', '이어폰']
 * randProduct({ language: 'en', count: 3 }); // ['Laptop', 'Keyboard', 'Earphone']
 */
export function randProduct(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Product words with the language and theme each one came from.
 *
 * @example
 * randProduct({ language: 'ko', output: 'detail' });
 * // [{ word: '노트북', language: 'ko', theme: 'product' }]
 */
export function randProduct(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randProduct(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('product', options);
}

File: packages/dart/lib/src/word/rand_product.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Things you buy — the manufactured end of the vocabulary.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.product] to `randWordDetails` — Dart has
/// no overloads, and seventeen more functions for it would be seventeen too many.
///
/// ```dart
/// randProduct(language: WordLanguage.ko, count: 3); // [노트북, 키보드, 이어폰]
/// randProduct(language: WordLanguage.en, count: 3); // [Laptop, Keyboard, Earphone]
/// ```
List<String> randProduct({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.product,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
